src/xToken.ts

```diff
@@ -642,13 +642,21 @@ export class XToken {
       return Promise.reject(new Error('Invalid user address'))
     }
 
-    return Promise.all([
-      getBalancerV2PortfolioItem(X_SNX_A, address, this.provider),
-      getBalancerPortfolioItem(X_AAVE_A, address, this.provider),
-      getBalancerPortfolioItem(X_AAVE_B, address, this.provider),
-      getKyberPortfolioItem(X_KNC_A, address, this.provider),
-      getBancorPortfolioItem(X_BNT_A, address, this.provider),
-    ])
+    const { chainId } = await this.provider.getNetwork()
+    switch (chainId) {
+      case ChainId.Mainnet:
+        return Promise.all([
+          getBalancerV2PortfolioItem(X_SNX_A, address, this.provider),
+          getBalancerPortfolioItem(X_AAVE_A, address, this.provider),
+          getBalancerPortfolioItem(X_AAVE_B, address, this.provider),
+          getKyberPortfolioItem(X_KNC_A, address, this.provider),
+          getBancorPortfolioItem(X_BNT_A, address, this.provider),
+        ])
+      case ChainId.Arbitrum:
+        return []
+      default:
+        return []
```

Comment:
We probably should be throwing an error here, if the user isn't connected to `Mainnet` or `Arbitrum` chain instead of returning empty array